File: examples/test_dataclass.py
"""Dataclass example for v1.2.0"""
from dataclasses import dataclass

@dataclass
class Point:
    """A 2D point dataclass"""
    x: int
    y: int
    
    def move_by(self, dx: int, dy: int):
        """Move point by delta"""
        self.x += dx
        self.y += dy
    
    def distance_to(self, other) -> float:
        """Calculate distance to another point"""
        dx = self.x - other.x
        dy = self.y - other.y
        return (dx * dx + dy * dy) ** 0.5
    
    @staticmethod
    def origin():
        """Create a point at origin"""
        return Point(0, 0)
    
    @classmethod
    def from_tuple(cls, coords: tuple[int, int]):
        """Create from tuple"""
        return cls(coords[0], coords[1])
    
    @property
    def magnitude(self) -> float:
        """Distance from origin"""
        return (self.x * self.x + self.y * self.y) ** 0.5

# Test function
def test_point():
    # Create points
    p1 = Point(0, 0)
    p2 = Point(3, 4)
    
    # Instance method
    p1.move_by(1, 1)
    
    # Static method
    origin = Point.origin()
    
    # Class method
    p3 = Point.from_tuple((5, 5))
    
    # Property access
    mag = p2.magnitude
    
    # Calculate distance
    dist = p1.distance_to(p2)
    
    return dist, mag

File: examples/test_dataclass.rs
use serde_json;
#[doc = "// NOTE: Map Python module 'dataclasses'(tracked in DEPYLER-0424)"]
#[derive(Debug, Clone, PartialEq)]
pub struct Point {
    pub x: i32,
    pub y: i32,
}
impl Point {
    pub fn new(x: i32, y: i32) -> Self {
        Self { x, y }
    }
    pub fn move_by(&mut self, dx: i32, dy: i32) {
        self.x = self.x + dx;
        self.y = self.y + dy;
    }
    pub fn distance_to(&self, other: serde_json::Value) -> f64 {
        let mut dx = self.x - other.x;
        let mut dy = self.y - other.y;
        return (dx * dx + dy * dy as f64).powf(0.5 as f64);
    }
    pub fn origin() {
        return Point::new(0, 0);
    }
    pub fn from_tuple(coords: (i32, i32)) {
        return Self::new(coords[0 as usize], coords[1 as usize]);
    }
    pub fn magnitude(&self) -> f64 {
        return (self.x * self.x + self.y * self.y as f64).powf(0.5 as f64);
    }
}
#[doc = r" Stub for local import from module: #module_name"]
#[doc = r" DEPYLER-0615: Generated to allow standalone compilation"]
#[allow(dead_code, unused_variables)]
pub fn dataclass<T: Default>(_args: impl std::any::Any) -> T {
    Default::default()
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_point() -> (serde_json::Value, serde_json::Value) {
    let mut p1 = Point::new(0, 0);
    let p2 = Point::new(3, 4);
    p1.move_by(1, 1);
    let mag = p2.magnitude;
    let dist = p1.distance_to(p2);
    (dist, mag)
}
